clients/drcachesim/scheduler/scheduler_dynamic.cpp

```diff
@@ -179,7 +186,11 @@ typename scheduler_tmpl_t<RecordType, ReaderType>::stream_status_t
 scheduler_dynamic_tmpl_t<RecordType, ReaderType>::swap_in_input(output_ordinal_t output,
                                                                 input_ordinal_t input)
 {
-    // TODO i#7067: Track peak live core usage per workload here.
+    if (input == sched_type_t::INVALID_INPUT_ORDINAL)
+        return sched_type_t::STATUS_OK;
+    workload_info_t &workload = workloads_[inputs_[input].workload];
+    if (workload.output_limit > 0)
+        workload.live_outputs->fetch_add(1, std::memory_order_release);
```

Comment:
Is it a fair expectation for this swap_in_input API that the derived classes shouldn't need to provide their own synchronization? (I know that that's not the case currently, but should it be made to be so)

Xref the other comment about races in and between swap_in_input and swap_in_output.

Reply:
> What kind of synchronization? Accessing shared structures has to be synchronized;

See other comment. If we want to handle races between swap_in_input and swap_in_output we may want to introduce our own synchronization which would make them thread-safe, obviating any need for the sub-class to add synch.

Reply:
It has to be up to the subclasses as they have very different concurrency and ordering behavior: see other comment.